airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/state/StateManagerFactory.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.relationaldb.state;
+
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.integrations.source.relationaldb.models.DbState;
+import io.airbyte.protocol.models.AirbyteStateMessage;
+import io.airbyte.protocol.models.AirbyteStateMessage.AirbyteStateType;
+import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Factory class that creates {@link StateManager} instances based on the provided state.
+ */
+public class StateManagerFactory {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(StateManagerFactory.class);
+
+  /**
+   * Private constructor to prevent direct instantiation.
+   */
+  private StateManagerFactory() {}
+
+  /**
+   * Creates a {@link StateManager} based on the provided state object and catalog. This method will
+   * handle the conversion of the provided state to match the requested state manager based on the
+   * provided {@link AirbyteStateType}.
+   *
+   * @param supportedStateType The type of state supported by the connector.
+   * @param initialState The deserialized initial state that will be provided to the selected
+   *        {@link StateManager}.
+   * @param catalog The {@link ConfiguredAirbyteCatalog} for the connector that will utilize the state
+   *        manager.
+   * @return A newly created {@link StateManager} implementation based on the provided state.
+   */
+  public static StateManager createStateManager(final AirbyteStateType supportedStateType,
+                                                final List<AirbyteStateMessage> initialState,
+                                                final ConfiguredAirbyteCatalog catalog) {
+    if (initialState != null && !initialState.isEmpty()) {
+      final AirbyteStateMessage airbyteStateMessage = initialState.get(0);
+      switch (supportedStateType) {
+        case LEGACY:
```

Comment:
@benmoriceau Not sure that I follow.  Because of the enum value, it has to be one of the three enumeration values.  I suppose that the type could be `null`, but that will cause a `NullPointerException`.